### PR TITLE
Update project README

### LINKED ISSUE #21824: Docs: Refresh README
Refresh the [README](https://github.com/wordpress-mobile/WordPress-iOS/blob/trunk/README.md) for the project.

```[tasklist]
### Tasks
- [x] Update cloning instructions to [include all methods](https://docs.github.com/en/repositories/creating-and-managing-repositories/cloning-a-repository)
- [x] Remove mention of "M1-based Mac support is coming"
- [x] Update Xcode version to be a minimum of [`.xcode-version`](https://github.com/wordpress-mobile/WordPress-iOS/blob/trunk/.xcode-version)
- [x] Remove outdated badges
- [x] Add "Website URL" field to setup instructions
```

### PR DESCRIPTION
**Pull request by twstokes**
Closes #21824

See the task list in https://github.com/wordpress-mobile/WordPress-iOS/issues/21824 for the changes made.

Removing the badges was an opinionated change, but my thoughts are:
- Statuses are easily accessible from the GitHub UI these days
- Status badges can break and are another thing to maintain

But am open to keeping them if we want.

### CODE REVIEW

**Comment by twstokes** [comment]
👋 Hey @mokagio - mind giving this one a look when convenient?

File: README.md [hunk 1]
```diff
@@ -1,19 +1,14 @@
...
 Please refer to the sections below for more detailed information. The instructions assume the work is performed from a command line.
 
-> Please note – these setup instructions only apply to Intel-based machines. M1-based Mac support is coming, but isn't yet supported by our tooling.
-
 ### Getting Started
 
-1. [Download](https://developer.apple.com/downloads/index.action) and install Xcode. *WordPress for iOS* requires Xcode 11.2.1 or newer.
-1. From a command line, run `git clone git@github.com:wordpress-mobile/WordPress-iOS.git` in the folder of your preference.
-1. Now, run `cd WordPress-iOS` to enter the working directory.
+1. [Download](https://developer.apple.com/downloads/index.action) and install Xcode. *WordPress for iOS* requires Xcode `14.3.1` or newer.
```
Comment (mokagio) [inline README.md]: A spelled out Xcode version in the README is bound to become out of sync, like the diff shows.

What do you think of mentioning the `.xcode-version` file?

https://github.com/wordpress-mobile/WordPress-iOS/blob/05a954e50ad82c1e651ce0e972bcb8dadc5c7be6/.xcode-version

```suggestion
1. [Download](https://developer.apple.com/downloads/index.action) and install Xcode. Refer to the `.xcode-version` file for the minimum required version.
```

File: README.md [hunk 2]
```diff
@@ -1,19 +1,14 @@
...
+1. [Clone this repository](https://docs.github.com/en/repositories/creating-and-managing-repositories/cloning-a-repository) in the folder of your preference.
+1. In a terminal, enter the `WordPress-iOS` directory where it was cloned.
```
Comment (mokagio) [inline README.md]: Is this line necessary?

From one point of view, it's good not to make assumptions. Then again, it should be a given that once you clone a repo, you go into that folder to work on it..

**Review by mokagio — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (twstokes) [reply README.md]: Good call! One less thing to maintain. 👍

File: README.md (hunk 2 above)
Comment (twstokes) [reply README.md]: Good call, tweaked in https://github.com/wordpress-mobile/WordPress-iOS/pull/21825/commits/3e7ceeca8dbd2e2dbb04497628926659d221297c.

At the risk of being verbose I was thinking about the users who clone via GUI tools and needed to be in the terminal at that point. 👍

**Pull request merged**